01_auth_api/src/db/index.ts

```diff
@@ -0,0 +1,25 @@
+import * as dotenv from 'dotenv';
+import {Pool, QueryConfig, QueryResult} from 'pg'
+import {QueryValues} from "./index.types";
+
+dotenv.config();
+
+const pool = new Pool();
+
+pool.on('error', (err: Error) => {
+    console.error('Database connection error:', err);
+});
+
+export const checkDatabaseConnection = async (): Promise<void> => {
+    try {
+        const result: QueryResult = await pool.query('SELECT NOW() as currentTime');
+        const currentTime = result.rows[0].currenttime;
```

Comment:
We can see that your value can be `null`, which can be tricky and cause a crash, I suggest using in this case [Optional Chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)

Cover with try/catch also good as I see.

Reply:
You are right, for some reason I neglected the fact that the result may not arrive when requesting, so I typed it and added the following check.

Type:
```typescript
export type CheckConnectionResult = {
    time: Date | null;
}
``` 
Check:
```typescript
        const result: QueryResult<CheckConnectionResult> = await pool.query('SELECT NOW() as time');
        const currentTime = result.rows[0].time;
        if (!currentTime) {
            return sendError('Database connection error');
        }
```

Reply:
Yep, good, thanks